openlibrary/plugins/openlibrary/js/ol.js

```diff
@@ -75,48 +79,263 @@ export function closePop(){
     });
 }
 
-export default function init(){
-    var $searchResults = $('header#header-bar .search-component ul.search-results');
-    var $searchInput = $('header#header-bar .search-component .search-bar-input input[type="text"]');
-    var cover_url = function(id) {
-        return `//covers.openlibrary.org/b/id/${id}-S.jpg`
-    };
-    // stores the state of the search result for resizing window
-    var instantSearchResultState = false;
-    var searchModes, searchModeDefault, defaultFacet, searchFacets, composeSearchUrl, marshalBookSearchQuery, renderInstantSearchResults, setFacet, setMode, setSearchMode, options, q, parts, enteredSearchMinimized, searchExpansionActivated, toggleSearchbar, renderInstantSearchResult, val, facet_value;
-
-    // searches should be cancelled if you click anywhere in the page
-    $('body').on('click', function () {
-        instantSearchResultState = false;
-        $searchResults.empty();
-    });
-    // but clicking search input should not empty search results.
-    $searchInput.on('click', false);
+const SEARCH_MODES = ['everything', 'ebooks', 'printdisabled'];
+const SEARCH_MODE_DEFAULT = 'ebooks';
+const SEARCH_FACETS = {
+    'title': 'books',
+    'author': 'authors',
+    'lists': 'lists',
+    'subject': 'subjects',
+    'all': 'all',
+    'advanced': 'advancedsearch',
+    'text': 'inside'
+};
+const DEFAULT_SEARCH_FACET = 'all';
+const RENDER_INSTANT_SEARCH_RESULT = {
+    books(work) {
+        const author_name = work.author_name ? work.author_name[0] : '';
+        $('header#header-bar .search-component ul.search-results').append(
+            `<li class="instant-result">
+                <a href="${work.key}">
+                    <img src="//covers.openlibrary.org/b/id/${work.cover_i}-S.jpg"/>
+                    <span class="book-desc">
+                        <div class="book-title">${work.title}</div> by <span class="book-author">${author_name}</span>
+                    </span>
+                </a>
+            </li>`
+        );
+    },
+    authors(author) {
+        // Todo: default author img to: https://dev.openlibrary.org/images/icons/avatar_author-lg.png
+        $('header#header-bar .search-component ul.search-results').append(
+            `<li>
+                <a href="/authors/${author.key}">
+                    <img src="http://covers.openlibrary.org/a/olid/${author.key}-S.jpg"/>
+                    <span class="author-desc"><div class="author-name">${author.name}</div></span>
+                </a>
+            </li>`
+        );
+    }
+}
 
-    $(window).scroll(function(){
-        var scroller = $('#formScroll');
-        if(isScrolledIntoView(scroller)){$('#scrollBtm').show();}else{$('#scrollBtm').hide();}
-    });
+/** Manages search state variables */
```

Comment:
Could you expand this? Why does it need to be a class? Is it an abstraction for localStorage?

Reply:
Have you considered pure functions instead? They're much easier to test!

Reply:
This whole SearchState thing is removed in a few commits; I couldn't remove it from the history in any resonable way, unfortunately.

Reply:
Which commit? Would it make sense to squash these 2? It's a big commit anyhow, so that might be easier and less confusing to review (I'm reviewing one at a time since there's so many)